src/CruxComponent.tsx

```diff
@@ -89,29 +89,29 @@ export class CruxComponentCreator {
             }
 
             showCreateModal = () => {
-                this.setState(Object.assign({}, this.state, { showCreateModal: true }))
```

Comment:
react internally does this, we can just path the changes in state object, it'll just patch given object, not replace it.

Reply:
Thanks for correcting this. This was my goofup and you will find this all through the codebase. I came to know about this only a couple of months back.